src/ariel-os-embassy/src/delegate.rs

```diff
@@ -30,43 +34,53 @@ use crate::sendcell::SendCell;
 ///   SOME_VALUE.with(|val| *val = 1).await;
 /// }
 /// ```
-///
-/// TODO: this is a PoC implementation.
-/// - takes 24b for each delegate (on arm), which seems too much.
-/// - doesn't protect at all against calling [`lend()`](Delegate::lend) or
-///   [`with()`](Delegate::with) multiple times
-///   each, breaking safety assumptions. So while the API seems OK, the implementation
-///   needs work.
+// TODO: this is a PoC implementation.
+// - Takes 25 B for each delegate (on arm), which seems too much.
 #[derive(Default)]
 pub struct Delegate<T> {
     send: Signal<CriticalSectionRawMutex, SendCell<*mut T>>,
     reply: Signal<CriticalSectionRawMutex, ()>,
+    was_exercised: AtomicBool,
 }
 
+impl<T> !Send for Delegate<T> {}
+
 impl<T> Delegate<T> {
     /// Creates a new [`Delegate`].
     pub const fn new() -> Self {
         Self {
             send: Signal::new(),
             reply: Signal::new(),
+            was_exercised: AtomicBool::new(false),
```

Comment:
maybe, if we turn this into a tristate enum (or `AtomicU8` with an enum wrapper to have it atomic), we can make the lend also safe?
(can do as follow up)